internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -1114,12 +1114,59 @@ func newAssistedCM(ctx context.Context, log logrus.FieldLogger, asc ASC) (client
 		copyEnv(cm.Data, "HTTP_PROXY")
 		copyEnv(cm.Data, "HTTPS_PROXY")
 		copyEnv(cm.Data, "NO_PROXY")
+		copyEnv(cm.Data, "DEPLOYMENT_TYPE")
+		copyEnv(cm.Data, "DEPLOYMENT_VERSION")
+		getDeploymentData(ctx, cm, asc)
 		return nil
 	}
 
 	return cm, mutateFn, nil
 }
 
+func getDeploymentData(ctx context.Context, cm *corev1.ConfigMap, asc ASC) {
+	if cm.Data["DEPLOYMENT_TYPE"] != "" && cm.Data["DEOLOYMENT_VERSION"] != "" {
+		return
+	}
+	//TODO: Remove this whole block and getVersionFromDeployment() function
+	// once ACM/MCE allows env var injection https://issues.redhat.com/browse/ACM-9362
+	deploy := &appsv1.Deployment{}
+	// Try getting the ACM deployment first
+	if err := asc.Client.Get(ctx, types.NamespacedName{Name: "multiclusterhub-operator", Namespace: "open-cluster-management"}, deploy); err == nil {
```

Comment:
What's the RBAC policy used for the running application? Did we try this in a real world scenario? There might be some RBAC limitations on fetching objects in other namespaces

Reply:
Good catch, I did test this out in a real world scenario (deployed OCP cluster using cluster bot then installed ACM via the operator hub) and it was able to query for the deployment alright, but I can look into RBAC generation to verify.

Reply:
Just checked and ACM does give us a clusterrole which allows us to see deployments in all namespaces
https://github.com/stolostron/backplane-operator/blob/57e760d08c6598915fc169d5c836e0f11c3af037/pkg/templates/charts/toggle/assisted-service/templates/assisted-service-clusterrole.yaml#L283-L286